distutils/dist.py

```diff
@@ -346,7 +346,12 @@ def _gen_paths(self):
         prefix = '.' * (os.name == 'posix')
         filename = prefix + 'pydistutils.cfg'
         if self.want_user_cfg:
-            yield pathlib.Path('~').expanduser() / filename
+            try:
+                yield pathlib.Path('~').expanduser() / filename
+            except RuntimeError:
+                warnings.warn(
+                    "Failed to locate user home directory. Skipping user config."
+                )
```

Comment:
I'm wondering if we even need a warning. This code is already branchy enough. I'm thinking maybe just suppress the error:

```suggestion
            with contextlib.suppress(RuntimeError):
                yield pathlib.Path('~').expanduser() / filename
```

That will also address the coverage issue.